src/utils.ts

```diff
@@ -1,15 +1,61 @@
 import { SelectionEntry, UIDisplayData } from '@aics/simularium-viewer';
 import { VisibilitySelectionMap } from './constants';
 
-export const getNewSelectionMap = (
+export const getNewMapAfterTopLevelCheckboxClick = (
   agentName: string,
   currentVisibilityMap: VisibilitySelectionMap
 ): VisibilitySelectionMap => {
   const newMap: VisibilitySelectionMap = { ...currentVisibilityMap };
-  if (currentVisibilityMap[agentName].length === 0) {
+  if (currentVisibilityMap[agentName].includes(agentName)) {
+    newMap[agentName] = [];
+  } else {
     newMap[agentName] = [agentName];
+  }
+  return newMap;
+};
+
+export const getNewMapAfterChildAgentClick = (
+  agentName: string,
+  childAgentName: string,
+  allChildren: string[],
```

Comment:
why does this function need both the agentName and allChildren? Isn't it able to get the children by the agent name?

Reply:
I pulled these functions out of the context file in response to feedback, so they lost access to the context as they are not in the component tree and not wrapped by the provider. That has driven some of the way this all gets written because the utils/selectors only know what is passed in to them.

Reply:
the one above does seem like a util function because it's just mapping over data and then returning the same thing every time. I see why you wanted these functions next to each other, but this one is not really like that one. There are a bunch of functions within this function that could be considered util functions